addon/utils/serializer-partial-model-handler.js

```diff
@@ -0,0 +1,33 @@
+export default {
+  preNormalize: function(modelClass, resourceHash, context, normalizePartialRelationshipStrategy) {
+    if (modelClass._isPartialModel) {
```

Comment:
Maybe move this up, so we don't enter this code path if no partial model handling